### ISSUE TITLE
Using alternate primary key types (such as UUID)

### ISSUE THREAD

**Issue by jwoertink** [luckyframework/lucky_record#246]
Sometimes you may want to use UUID or a custom string for your `ID` column as your primary key. Right now if you try to override this, lucky blows up because of [this](https://github.com/luckyframework/lucky_record/blob/master/src/lucky_record/model.cr#L13). Having a way to customize this would be pretty sweet.

In my case, I have an app that has to run in parallel with another app. The other app uses integer IDs. When a record comes in like `/sites/1`, I know this is data from the other app. When it comes in like `/sites/abc`, I know it's from this app. Using a UUID as an ID can also help to avoid people looking in to how many records you may have, or doing a hack like trying to access a record by incrementing the id in a route.

**Comment by sfcgeorge** [comment]
Rails 5.1 allows configuring primary key type. By default it's still a number but [now supports UUID](https://lab.io/articles/2017/04/13/uuids-rails-5-1/). Apart from the metadata leaking point above, UUIDs also potentially allow sharding and whatnot.

Rails also changed the default numerical primary key type from SQL "integer" to "bigint". Because if you have something like a log table, it is possible to exhaust all integers then all of a sudden everything breaks (a company experienced this, which prompted the change). So the default should at least be changed from Int32 to Int64.

**Comment by paulcsmith** [comment]
👍 LuckyRecord should definitely support UUID and setting a default of UUID would be great too.

I'm imaging that for one-off tables it would look like this:

```
table :users, primary_key: :uuid do
end

# Or if you set it to default to :uuid
table :users, primary_key: :bigint do
end
```

I think making it a default is a bit trickier so should be a separate issue, but making it configurable in the `table` macro should be fairly straightforward.
